### PR TITLE
Visualisation UI updates

### PR DESCRIPTION
**Pull request by mmilaprat**
related with issue: policycompass/policycompass#453

In this PR we update visulisation:

Legends ( add dataset title to the individual (always), Add Events in legends, Plot an individual legend per line)
Revert button anly in edit mode and enable when user update form
Improve modal windows (link evends and datasets)
Paginations in modal windows
Map button next to line, pie and bars button

### CODE REVIEW

**Comment by mmilaprat** [comment]
If a line chart (in edit mode) has releated events and the visualisation granularity is dfferent from day, users can select where to plot dots (beggin, middle or end of the period).

**Comment by cbotsikas** [comment]
@mmilaprat please split such changes in smaller PRs in the future. Evaluating PRs like this, is hard to follow the changes and requires time..
